Instagram/View Controllers/LoginViewController.m

```diff
@@ -12,57 +12,88 @@ @interface LoginViewController ()
 
 @property (weak, nonatomic) IBOutlet UITextField *usernameField;
 @property (weak, nonatomic) IBOutlet UITextField *passwordField;
+@property (weak, nonatomic) IBOutlet UIActivityIndicatorView *activityIndicator;
+@property (strong, nonatomic) UIAlertController *blankAlert;
+@property (strong, nonatomic) UIAlertController *registrationAlert;
+@property (strong, nonatomic) UIAlertController *loginAlert;
 
 @end
 
 @implementation LoginViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
+    
+    self.blankAlert = [UIAlertController alertControllerWithTitle:@"Username or password is empty." message:@"Please try again!" preferredStyle:(UIAlertControllerStyleAlert)];
```

Comment:
This is interesting. A bit unusual. Typically you'll see a method to produce these on demand like, 
```
- (UIAlertController *)alertControllerForError:(MyError *)error
```

They're not expensive to create up-front, just maybe a little messy to have all this in viewDidLoad.